cerberus/cerberus.py

```diff
@@ -208,6 +208,36 @@ def _validate(self, document, schema=None, update=False, context=None):
         elif self.schema is None:
             raise SchemaError(errors.ERROR_SCHEMA_MISSING)
 
+
+        # if self.schema is a list, then we want to validate the document against each item
+        # in the list and return true if one schema in the list validates.
+        if isinstance( self.schema, Sequence ) and not isinstance(self.schema, _str_type):
+            # assume that we no schemas validate
+            valid = False
+            # keep an error stack to pass on in case no schemas validate
+            errorstack = []
+            # for each schema...
+            for s in self.schema:
+                validator = self.__get_child_validator( schema=s )
+                validator.validate(document
+                                 , context=context
+                                 , update=update)
+                if validator.errors:
+                    # there were errors. append to the list
+                    errorstack.append( validator.errors )
+                else:
+                    # no errors occured, set valid and get out of here
+                    valid = True
+                    break
+
+            if not valid:
+                # no schema in the list validated.
+                # add all the errors we have collected to the errors list
+                for i in xrange(len(errorstack)):
+                    self._error( "schema %d"%i, errorstack[i] )
+
+            return valid
```

Comment:
Well, that's embarrassing, the only test that fails is the test I wrote to demonstrate it work! Let me look into it. This may be related to [107](https://github.com/nicolaiarocci/cerberus/issues/107). I am using a patch that is not in the tree because I see the same error that was reported but funkyfuture does not.